clients/spark/core/src/test/scala/io/treeverse/clients/StorageUtilsSpec.scala

```diff
@@ -0,0 +1,146 @@
+        awsS3ClientBuilder,
+        US_WEST_2,
+        BUCKET_NAME
+      )
+
+      server.getRequestCount should equal(1)
+      val request: RecordedRequest = server.takeRequest()
+      initializedClient should not be null
+      initializedClient.getRegion.toString should equal(US_WEST_2)
+      extractBucketFromRecordedRequest(request) should equal(BUCKET_NAME)
+      request.getMethod should equal(HttpMethod.HEAD.toString)
+    }
+
+    it("should create the S3 client successfully with US STANDARD region validation") {
+      server.enqueue(new MockResponse().setBody("successful validation response"))
+      val initializedClient: AmazonS3 = initializeClient()
+
+      server.getRequestCount should equal(1)
+
+      val request: RecordedRequest = server.takeRequest()
+      initializedClient should not be null
+      initializedClient.getRegion.name should equal(Region.US_Standard.name())
+      extractBucketFromRecordedRequest(request) should equal(BUCKET_NAME)
+      request.getMethod should equal(HttpMethod.HEAD.toString)
+    }
+
+    it("should get the correct region for the given bucket and create the S3 client successfully") {
+      server.enqueue(new MockResponse().setResponseCode(HttpStatus.SC_FORBIDDEN))
+      server.enqueue(
+        new MockResponse()
+          .setBody(generateGetBucketLocationResponseWithRegion(US_WEST_2))
+          .setResponseCode(HttpStatus.SC_OK)
+      )
+      val initializedClient: AmazonS3 = initializeClient()
+
+      server.getRequestCount should equal(2)
+      val headBucketRequest: RecordedRequest = server.takeRequest()
+      val getLocationRequest: RecordedRequest = server.takeRequest()
+      initializedClient should not be null
+      initializedClient.getRegion.toString should equal(US_WEST_2)
+      extractBucketFromRecordedRequest(headBucketRequest) should equal(BUCKET_NAME)
+      headBucketRequest.getMethod should equal(HttpMethod.HEAD.toString)
+      getLocationRequest.getMethod should equal(HttpMethod.GET.toString)
+    }
+
+    it("should fail creating a client due to failed 'getBucketLocation' request") {
+      server.enqueue(new MockResponse().setResponseCode(HttpStatus.SC_FORBIDDEN))
+      server.enqueue(
+        new MockResponse()
+          .setResponseCode(HttpStatus.SC_NOT_FOUND)
+      )
+
+      assertThrows[SdkClientException] {
+        initializeClient()
+      }
+
+      server.getRequestCount should equal(2)
+      val headBucketRequest: RecordedRequest = server.takeRequest()
+      val getLocationRequest: RecordedRequest = server.takeRequest()
+      extractBucketFromRecordedRequest(headBucketRequest) should equal(BUCKET_NAME)
+      headBucketRequest.getMethod should equal(HttpMethod.HEAD.toString)
+      getLocationRequest.getMethod should equal(HttpMethod.GET.toString)
+    }
+  }
+
+  private def extractBucketFromRecordedRequest(request: RecordedRequest): String = {
+    val splicedRequestLine = request.getRequestLine.split('/')
```

Comment:
```suggestion
    val splitedRequestLine = request.getRequestLine.split('/')
```
Is that what you meant?

Reply:
```suggestion
    val splitRequestLine = request.getRequestLine.split('/')
```
It should actually be this... Probably fell into IntelliJ's spelling correction trap...
Thanks!